pkg/cli/alpha/internal/update/update.go

```diff
@@ -0,0 +1,372 @@
+// to ensure all the necessary components are generated, formatted and linted.
+func runMakeTargets() {
+	targets := []string{"manifests", "generate", "fmt", "vet", "lint-fix"}
+	for _, target := range targets {
+		err := util.RunCmd(fmt.Sprintf("Running make %s", target), "make", target)
+		if err != nil {
+			log.Warnf("make %s failed: %v", target, err)
+		}
+	}
+}
+
+// runAlphaGenerate executes the old Kubebuilder version's 'alpha generate' command
+// to create clean scaffolding in the ancestor branch. This uses the downloaded
+// binary with the original PROJECT file to recreate the project's initial state.
+func runAlphaGenerate(tempDir, version string) error {
+	log.Infof("Generating project with version %s", version)
+	// Temporarily modify PATH to use the downloaded Kubebuilder binary
+	tempBinaryPath := tempDir + "/kubebuilder"
+	originalPath := os.Getenv("PATH")
+	tempEnvPath := tempDir + ":" + originalPath
+
+	if err := os.Setenv("PATH", tempEnvPath); err != nil {
+		return fmt.Errorf("failed to set temporary PATH: %w", err)
+	}
+
+	defer func() {
+		if err := os.Setenv("PATH", originalPath); err != nil {
+			log.Errorf("failed to restore original PATH: %v", err)
+		}
+	}()
+
+	// TODO: we need improve the implementation from utils to allow us
+	// to pass the path of the binary and use it to run the alpha generate command.
+	cmd := exec.Command(tempBinaryPath, "alpha", "generate")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Env = os.Environ()
+
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to run alpha generate: %w", err)
+	}
+	log.Info("Successfully ran alpha generate", version)
+
+	// TODO: Analyse if this command is still needed in the future.
+	// It was added because the alpha generate command in versions prior to v4.7.0 does
+	// not run those commands automatically which will not allow we properly ensure
+	// that all manifests, code generation, formatting, and linting are applied to
+	// properly do the 3-way merge.
+	runMakeTargets()
+	return nil
+}
+
+// prepareOriginalBranch creates the 'original' branch from ancestor and
+// populates it with the user's actual project content from the default branch.
+// This represents the current state of the user's project.
+func (opts *Update) prepareOriginalBranch() error {
+	gitCmd := exec.Command("git", "checkout", "-b", opts.OriginalBranch)
+	if err := gitCmd.Run(); err != nil {
+		return fmt.Errorf("failed to checkout branch %s: %w", opts.OriginalBranch, err)
+	}
+	log.Infof("Successfully checked out %s", opts.OriginalBranch)
+
+	gitCmd = exec.Command("git", "checkout", opts.FromBranch, "--", ".")
+	if err := gitCmd.Run(); err != nil {
+		return fmt.Errorf("failed to checkout content from %s branch onto %s: %w", opts.FromBranch, opts.OriginalBranch, err)
+	}
+	log.Infof("Successfully checked out content from %s onto %s branch", opts.FromBranch, opts.OriginalBranch)
+
+	gitCmd = exec.Command("git", "add", "--all")
```

Comment:
@vitorfloriano we need to use add --all to ensure that all files will be added 
we cannot use git add . , since it can leave files behind